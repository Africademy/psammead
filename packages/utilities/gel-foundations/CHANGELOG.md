# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.0   | [PR#XX](https://github.com/BBC-News/psammead/pull/XX) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |
