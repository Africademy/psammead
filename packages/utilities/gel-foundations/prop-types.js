module.exports = require('./dist/prop-types'); // eslint-disable-line import/no-unresolved
