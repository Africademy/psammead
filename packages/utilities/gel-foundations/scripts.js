/* eslint-disable global-require */
module.exports = {
  news: require('./dist/scripts/news').default,
  arabic: require('./dist/scripts/arabic').default,
  burmese: require('./dist/scripts/burmese').default,
  cjkThai: require('./dist/scripts/cjk_thai').default,
  hindiNepaliBengali: require('./dist/scripts/hindi_nepali_bengali').default,
  latin: require('./dist/scripts/arabic').default,
  mundo: require('./dist/scripts/arabic').default,
  sinhala: require('./dist/scripts/sinhala').default,
  tamil: require('./dist/scripts/tamil').default,
};
