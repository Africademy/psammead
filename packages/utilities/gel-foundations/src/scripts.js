/* eslint-disable global-require */
module.exports = {
  latin: require('../dist/scripts/latin-and-cyrillic').default,
  cyrillic: require('../dist/scripts/latin-and-cyrillic').default,
  latinDiacritics: require('../dist/scripts/latin-with-diacritics').default,
  arabic: require('../dist/scripts/arabic').default,
  arabicPashto: require('../dist/scripts/arabic-pashto').default,
  bengali: require('../dist/scripts/bengali').default,
  burmese: require('../dist/scripts/burmese').default,
  hindi: require('../dist/scripts/devanagari-and-gurmukhi').default,
  nepali: require('../dist/scripts/devanagari-and-gurmukhi').default,
  sinhalese: require('../dist/scripts/sinhalese').default,
  tamil: require('../dist/scripts/tamil').default,
  thai: require('../dist/scripts/thai').default,
  chinese: require('../dist/scripts/no-ascenders-or-descenders').default,
  korean: require('../dist/scripts/no-ascenders-or-descenders').default,
  ethiopic: require('../dist/scripts/no-ascenders-or-descenders').default,
};
