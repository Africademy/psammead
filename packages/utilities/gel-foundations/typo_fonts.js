module.exports = require('./dist/typo_fonts'); // eslint-disable-line import/no-unresolved
