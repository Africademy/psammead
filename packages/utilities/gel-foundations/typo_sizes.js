module.exports = require('./dist/typo_sizes'); // eslint-disable-line import/no-unresolved
