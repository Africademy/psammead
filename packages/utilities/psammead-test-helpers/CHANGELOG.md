# Psammead Test Helpers Changelog

| Version | Description |
|---------|-------------|
| 0.1.2   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
