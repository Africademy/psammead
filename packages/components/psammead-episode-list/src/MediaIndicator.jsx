import React from 'react';
import styled, { css } from 'styled-components';
import { string } from 'prop-types';

const Wrapper = styled.div`
  display: inline-block;
  ${({ size }) => css`
    width: ${size};
    height: ${size};
  `}
`;

const MediaIndicator = ({ size }) => (
  <Wrapper
    size={size}
    dangerouslySetInnerHTML={{
      __html: `
      <svg class="rounded-play-button" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 40">
          <path class="rounded-play-button__outer-circle" d="M20,0C8.954,0,0,8.954,0,20s8.954,20,20,20s20-8.954,20-20S31.046,0,20,0z"/>
          <path class="rounded-play-button__inner-circle" fill="#fff" d="M20,1.765C9.929,1.765,1.765,9.929,1.765,20S9.929,38.235,20,38.235S38.235,30.071,38.235,20
          S30.071,1.765,20,1.765z"/>
          <polygon class="rounded-play-button__triangle" points="15.799,26.94 27.062,20.009 15.799,13.078 "/>
      </svg>
      `,
    }}
  />
);

MediaIndicator.propTypes = {
  size: string.isRequired,
};

export default MediaIndicator;
