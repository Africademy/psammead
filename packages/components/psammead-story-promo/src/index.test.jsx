import React, { Fragment } from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { latin } from '@bbc/gel-foundations/scripts';
import MediaIndicator from '@bbc/psammead-media-indicator';
import StoryPromo, { Headline, Summary, Link } from './index';

const Image = <img src="https://foobar.com/image.png" alt="Alt text" />;

const Info = (
  <Fragment>
    <Headline script={latin}>
      <Link href="https://www.bbc.co.uk/news">The headline of the promo</Link>
    </Headline>
    <Summary script={latin}>The summary of the promo</Summary>
    <time>12 March 2019</time>
  </Fragment>
);

const mediaInfo = (
  <MediaIndicator
    duration="2:15"
    datetime="PT2M15S"
    offscreenText="Video 2 minutes 15 seconds"
  />
);

describe('StoryPromo', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <StoryPromo image={Image} info={Info} />,
  );
});

describe('StoryPromo with Media Indicator', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <StoryPromo image={Image} info={Info} mediaIndicator={mediaInfo} />,
  );
});

describe('StoryPromo - Top Story', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <StoryPromo image={Image} info={Info} topStory={true}/>,
  );
});

describe('StoryPromo - Top Story with Media Indicator', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <StoryPromo
      image={Image}
      info={Info}
      mediaIndicator={mediaInfo}
      topStory={true}
    />,
  );
});
