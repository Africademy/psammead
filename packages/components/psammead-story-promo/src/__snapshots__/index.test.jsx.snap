// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryPromo should render correctly 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(6,1fr);
  grid-column-gap: 0.5rem;
}

.c1 {
  grid-column: 1 / span 2;
}

.c2 {
  grid-column: 3 / span 4;
}

.c3 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #3F3F42;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
  font-weight: 700;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #3F3F42;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
}

@media (min-width:37.5rem) {
  .c0 {
    grid-column-gap: 1rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-template-columns: repeat(12,1fr);
  }
}

@media (min-width:80rem) {
  .c1 {
    grid-column: 1 / span 4;
  }
}

@media (min-width:80rem) {
  .c2 {
    grid-column: 5 / span 8;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:80rem) {
  .c3 {
    grid-template-columns: repeat(12,1fr);
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      alt="Alt text"
      src="https://foobar.com/image.png"
    />
  </div>
  <div
    className="c2"
  >
    <h3
      className="c3"
    >
      The headline of the promo
    </h3>
    <p
      className="c4"
    >
      The summary of the promo
    </p>
    <time>
      12 March 2019
    </time>
  </div>
</div>
`;
